### ISSUE TITLE
How to get secret

### ISSUE THREAD

**Issue by ntkernel** [TwizzyIndy/hw_atcmdserver_xcode#1]
Hello,i get the response from my device"AT xxxx = XXXXXXXXXXXXXXXXXXXXXXXXX"(very long),but i don't know how to get the private key or decrypt it.Thank you.

**Comment by ntkernel** [comment]
when i use python to encrypt the seed,is the public key 
0912C0F87E7EFEE5D85858B316A3AB072F3131EF383E1A01F098E38AFB1686693F7A18A9CDA8A41A291D07500B84BDB37D8B98C624D84ED242D306CE4A204BD9B8CD3668802854D48BE2D25D10625C739E4069CB7378EEC0F69185E8B02A186E8096E4EC2826CF06DDF82638409B4FF503A752E7B84F29728E9D3E34FA0551AF
i need to use?

**Comment by TwizzyIndy** [comment]
did you check out my post on [here](https://twizzyindy.github.io/2019/01/26/Sneaking-into-Huawei-Bootloader-And-ATCMDSERVER/)

**Comment by ntkernel** [comment]
sorry,i read your post but i dont know how to get the private key

**Comment by ntkernel** [comment]
i know the public key

import serial  
import time  
import base64
from Crypto.PublicKey import RSA
from Crypto.Cipher import PKCS1_v1_5 as Cipher_pkcsl_v1_5
import rsa
i1 = "af5105fa343e9d8e72294fb8e752a703f54f9b403826f8dd06cf2628ece496806e182ab0e88591f6c0ee7873cb69409e735c62105dd2e28bd45428806836cdb8d94b204ace06d342d24ed824c6988b7db3bd840b50071d281aa4a8cda9187a3f698616fb8ae398f0011a3e38ef31312f07aba316b35858d8e5fe7e7ef8c01209"
i2 = "0912C0F87E7EFEE5D85858B316A3AB072F3131EF383E1A01F098E38AFB1686693F7A18A9CDA8A41A291D07500B84BDB37D8B98C624D84ED242D306CE4A204BD9B8CD3668802854D48BE2D25D10625C739E4069CB7378EEC0F69185E8B02A186E8096E4EC2826CF06DDF82638409B4FF503A752E7B84F29728E9D3E34FA0551AF"
n1 = int(i1,16)
n2 = int(i2,16)
e = 65537
ok1 = rsa.PublicKey(n1,e)
ok2 = rsa.PublicKey(n2,e)


is the second one the public key?


i get the following string from my device


"83d8fda633b2448362823772ceafbd8bb64ca1d3dfd1f95066c8549184371dca93ff9cc50aa263e168493c241530a18531e9a2cfb2bb8f148875f96ca4758573a61a61c8135635f12270aae6a359c0ac47d4d6d86bcd70437594199369f3e4640babe87b15194d9e70af486ebfdb3886df5ba03edc55d6808b9b99548948df1fa825e6432660ffde35cc255e45c7a5dbe7b93eaeb8c88e0e423ae2d92fcc5315f405106dbca0cf03b49fc1027b9797c64a915c03e163fb338aebbd5be9e1374cb8368bb5776c304dbd679d7ac552670800dc07429c0b9814ccc7a17f4c012ec9a1c4f011d9915aa5bc24cca2f6a546328688e018301ed9a528bc123bd48a5a85"

and what should i do

**Comment by TwizzyIndy** [comment]
second one is Public Modulus (N).
I cant calculate or hack the private key yet.
so I cant implement yet. But we even got four ..
M = the message(128Bytes) which isnt encrypted (e.g: 0x0001FFFFFFFF..[random four bytes generated from /dev/random] FF) ( you can catch it using Frida script I made )
N = Modulus existed in g_table.bin
E = exponent we already know
C = Encrypted bytes out from device .. (e.g : AT^CHECKAUTHORITY=XX, i think its like ur third one )

We need to know the two prime factor .. p and q for calculate d

**Comment by ntkernel** [comment]
i get a 512 byte string from my device(256byte),i think my device's atcmdserver is different from others.
like this

83d8fda633b2448362823772ceafbd8bb64ca1d3dfd1f95066c8549184371dca93ff9cc50aa263e168493c241530a18531e9a2cfb2bb8f148875f96ca4758573a61a61c8135635f12270aae6a359c0ac47d4d6d86bcd70437594199369f3e4640babe87b15194d9e70af486ebfdb3886df5ba03edc55d6808b9b99548948df1fa825e6432660ffde35cc255e45c7a5dbe7b93eaeb8c88e0e423ae2d92fcc5315f405106dbca0cf03b49fc1027b9797c64a915c03e163fb338aebbd5be9e1374cb8368bb5776c304dbd679d7ac552670800dc07429c0b9814ccc7a17f4c012ec9a1c4f011d9915aa5bc24cca2f6a546328688e018301ed9a528bc123bd48a5a85

**Comment by TwizzyIndy** [comment]
yes its basically 256bytes coz its Hex String. You have to convert Hex2Bin u will got 128bytes cypher. 1 byte = 2 Hex values

**Comment by TwizzyIndy** [comment]
yes its fixed to RSA-2048 on later version i heard.
when i was wrote this script, it still 128bytes.
[https://github.com/TwizzyIndy/NaughtyPythonScripts/blob/master/HookAtcmd_android.py](url)

**Comment by ntkernel** [comment]
oo,I can't get the atcmdserver from my device,i can't unzip the vendor image

**Comment by kasnria001** [comment]
> yes its fixed to RSA-2048 on later version i heard. when i was wrote this script, it still 128bytes. [https://github.com/TwizzyIndy/NaughtyPythonScripts/blob/master/HookAtcmd_android.py](url)

What's your blog post?
